warehouse/oidc/utils.py

```diff
@@ -91,3 +94,9 @@ def find_publisher_by_issuer(session, issuer_url, signed_claims, *, pending=Fals
     else:
         # Unreachable; same logic error as above.
         return None  # pragma: no cover
+
+
+@dataclass
+class OIDCContext:
```

Comment:
Here is okay, although IMO `oidc/security_policy.py` (i.e., a new module) would also be acceptable. But given that it'd be the only member, here is probably better.

(Let's add a docstring explaining what this class is with some references to other APIs that use it; it's a little bit incongruous otherwise.)